src/script/api/script_admin.cpp

```diff
@@ -28,89 +28,73 @@
 			SQInteger res;
 			sq_getinteger(vm, index, &res);
 
-			data = fmt::format("{}", res);
+			json = res;
 			return true;
 		}
 
 		case OT_STRING: {
 			const SQChar *buf;
 			sq_getstring(vm, index, &buf);
 
-			size_t len = strlen(buf) + 1;
-			if (len >= 255) {
-				ScriptLog::Error("Maximum string length is 254 chars. No data sent.");
-				return false;
-			}
-
-			data = fmt::format("\"{}\"", buf);
+			json = std::string(buf);
 			return true;
 		}
 
 		case OT_ARRAY: {
-			data = "[ ";
+			json = nlohmann::json::array();
 
-			bool first = true;
 			sq_pushnull(vm);
 			while (SQ_SUCCEEDED(sq_next(vm, index - 1))) {
-				if (!first) data += ", ";
-				if (first) first = false;
+				nlohmann::json tmp;
 
-				std::string tmp;
-
-				bool res = MakeJSON(vm, -1, max_depth - 1, tmp);
+				bool res = MakeJSON(tmp, vm, -1, depth + 1);
 				sq_pop(vm, 2);
 				if (!res) {
 					sq_pop(vm, 1);
 					return false;
 				}
-				data += tmp;
+
+				json.push_back(tmp);
 			}
 			sq_pop(vm, 1);
-			data += " ]";
 			return true;
 		}
 
 		case OT_TABLE: {
-			data = "{ ";
+			json = nlohmann::json::object();
 
-			bool first = true;
 			sq_pushnull(vm);
 			while (SQ_SUCCEEDED(sq_next(vm, index - 1))) {
-				if (!first) data += ", ";
-				if (first) first = false;
-
-				std::string key;
-				std::string value;
-
-				/* Store the key + value */
-				bool res = MakeJSON(vm, -2, max_depth - 1, key) && MakeJSON(vm, -1, max_depth - 1, value);
+				/* Squirrel ensure the key is a string. */
+				assert(sq_gettype(vm, -2) == OT_STRING);
```

Comment:
Safe to assume this assert can't be triggered by user scripts?

Reply:
Yeah, that is what this comment tries to say. A table/dict in Squirrel always has strings as keys (by definition). So there is no need here to check other types. But I did wanted to add the assert to make sure the claim Squirrel makes it actually correct :)